### PR TITLE
US139032 Copy to Other Courses - Save and Copy button

### PR DESCRIPTION
**Pull request by jguann**
[US139032 ](https://rally1.rallydev.com/#/431372673576ud/iterationstatus?detail=%2Fuserstory%2F636555822723)

- Show 'Save and Copy' button on assignment creation if feature flag and config are on (https://github.com/Brightspace/lms/pull/23418)
- Clicking this button will save and close the assignment
- Another PR will be made for launching the Copy to Other Courses dialog at this point

Screenshot 2022-05-19 120145

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-editor-buttons.js [hunk 1]
```diff
@@ -74,6 +83,17 @@ class ActivityEditorButtons extends RtlMixin(LocalizeActivityEditorMixin(LitElem
...
 		this.dispatchEvent(event);
 	}
+	_saveAndCopy() {
+		const event = new CustomEvent('d2l-activity-editor-save-and-copy', {
+			detail: {
+				saveInPlace: false
+			},
+			bubbles: true,
+			composed: true,
+			cancelable: true
```
Comment (jorge-ramirez-arredondo) [inline components/d2l-activity-editor/d2l-activity-editor-buttons.js:93]: I don't think you need this to be cancelable. I don't know why the other ones are, but it doesn't look like any of these are cancelling the event.

**Comment by jorge-ramirez-arredondo** [comment]
It looks like you're supposed to provide translations for every lang file when you add a new term? Not sure if you're supposed to just copy it in English, or if there's something else you're expected to do here.

image

**Comment by jguann** [comment]
> It looks like you're supposed to provide translations for every lang file when you add a new term? Not sure if you're supposed to just copy it in English, or if there's something else you're expected to do here.
> 
> image

Yeah, I have to figure that out

**Comment by jorge-ramirez-arredondo** [comment]
I think you're just supposed to do fill in the lang terms in the other files just in English. At least, that's what was done in this other PR: https://github.com/BrightspaceHypermediaComponents/activities/pull/2614

File: components/d2l-activity-editor/d2l-activity-editor-buttons.js [hunk 2]
```diff
@@ -74,6 +83,17 @@ class ActivityEditorButtons extends RtlMixin(LocalizeActivityEditorMixin(LitElem
...
 		this.dispatchEvent(event);
 	}
+	_saveAndCopy() {
+		const event = new CustomEvent('d2l-activity-editor-save-and-copy', {
+			detail: {
+				saveInPlace: false
+			},
+			bubbles: true,
+			composed: true,
+			cancelable: false
```
Comment (jorge-ramirez-arredondo) [inline components/d2l-activity-editor/d2l-activity-editor-buttons.js:93]: Btw, this is totally fine, but `cancelable` will default to `false` if you don't specify it. So, you could also just leave it out.

**Review by jorge-ramirez-arredondo — APPROVED** [review]
Not sure if you need to wait for code owners to approve here, but it looks good to me at least.

**Pull request merged**